### PR TITLE
Vector Pool

### PR DESCRIPTION
**Pull request by pollend**
This commit adds a vector pool to TeraMath based of the changes provided by Adrijaned. A set of unit test still need to be built to verify if the pool is working correctly.

This commit is based off of #52

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -229,7 +229,7 @@ ext {
...
+                    <option name="notice" value="Copyright 2018 MovingBlocks&#10;&#10;Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);&#10;you may not use this file except in compliance with the License.&#10;You may obtain a copy of the License at&#10;&#10;     http://www.apache.org/licenses/LICENSE-2.0&#10;&#10;Unless required by applicable law or agreed to in writing, software&#10;distributed under the License is distributed on an &quot;AS IS&quot; BASIS,&#10;WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.&#10;See the License for the specific language governing permissions and&#10;limitations under the License." />
```
Comment (syntaxi) [inline build.gradle]: If we are going to update the build.gradle to fix the copywrite year we should probably copy the engine's method to automatically grab the current year.
See https://github.com/MovingBlocks/Terasology/blob/develop/config/gradle/ide.gradle#L38

**Review by syntaxi — COMMENTED** [review]
Only a brief look.

**Comment by syntaxi** [comment]
If this PR makes #52 redundant we should close that one.

**Comment by Adrijaned** [comment]
I'm afraid that either VectorPoolHelper has to stay and be filled with callers for all the possible types, or all the public methods have to be made static. That's something we've argued for several days with vampcat, and the helper way was his. Also, VectorPools shouldn't probably go into `geom`.

File: src/main/java/org/terasology/math/geom/VectorPool.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.math.geom;
+
+/**
+ * Pool of Vectors, or, optionally, any object.
```
Comment (Adrijaned) [inline src/main/java/org/terasology/math/geom/VectorPool.java]: Might be better not to suggest using VectorPools for pooling things that are not vectors.

File: src/generator/resources/VectorPool.st [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        if (free) {
+            pool.add(v);
+            return null;
+        }
+        if (pool.size() > 0) {
+            return pool.remove(pool.size() - 1);
+        }
+        return new $vectorpool/baseType()$();
+    }
+}
```
Comment (Adrijaned) [inline src/generator/resources/VectorPool.st]: missing newline

**Pull request merged**